modules/bg_controller/templates/pr-created-buildspec-source.yml.tpl

```diff
@@ -19,7 +19,7 @@ phases:
       - aws s3api delete-object --bucket s3-codepipeline-${app_name}-${env_type} --key ${env_name}-blue/source_artifacts.zip
       - head=$(echo $CODEBUILD_WEBHOOK_HEAD_REF | sed 's/origin\///' | sed 's/refs\///' | sed 's/heads\///')
       - base=$(echo $CODEBUILD_WEBHOOK_BASE_REF | sed 's/origin\///' | sed 's/refs\///' | sed 's/heads\///')
-      - aws ssm put-parameter --name /infra/${app_name}-${env_name}/merge_details --value  '[]' --type String --overwrite
+      - aws dynamodb delete-item --table-name MergeWaiter-${app_name}-${env_type} --key '{"APPLICATION" :{"S":"${app_name}-${env_name}"}}' || echo "not found"
```

Comment:
Wrap this with a retry (i.e. 3 retries), and if all the consumed retries failed, please write to log with the query against dynamodb.

Reply:
We do not do retries in build specs, this can get stack forever !